nuxeo-importer-stream/src/main/java/org/nuxeo/importer/stream/automation/FileBlobProducers.java

```diff
@@ -0,0 +1,148 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Benoit Delbosc
+ */
+package org.nuxeo.importer.stream.automation;
+
+import static org.nuxeo.importer.stream.automation.BlobConsumers.DEFAULT_LOG_CONFIG;
+
+import java.io.File;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.ecm.automation.OperationContext;
+import org.nuxeo.ecm.automation.core.Constants;
+import org.nuxeo.ecm.automation.core.annotations.Context;
+import org.nuxeo.ecm.automation.core.annotations.Operation;
+import org.nuxeo.ecm.automation.core.annotations.OperationMethod;
+import org.nuxeo.ecm.automation.core.annotations.Param;
+import org.nuxeo.ecm.core.api.NuxeoPrincipal;
+import org.nuxeo.importer.stream.message.BlobMessage;
+import org.nuxeo.importer.stream.producer.FileBlobMessageProducerFactory;
+import org.nuxeo.lib.stream.log.LogManager;
+import org.nuxeo.lib.stream.pattern.producer.ProducerPool;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.stream.StreamService;
+
+/**
+ * Create blob messages using a list of files.
+ *
+ * @since 10.2
+ */
+@Operation(id = FileBlobProducers.ID, category = Constants.CAT_SERVICES, label = "Produces blobs from a list of files", since = "10.2", description = "Produces blobs from a list of files.")
+public class FileBlobProducers {
+    private static final Log log = LogFactory.getLog(FileBlobProducers.class);
+
+    public static final String ID = "StreamImporter.runFileBlobProducers";
+
+    public static final String DEFAULT_BLOB_LOG_NAME = "import-blob";
+
+    @Context
+    protected OperationContext ctx;
+
+    @Param(name = "listFile")
+    protected String listFile;
+
+    @Param(name = "nbBlobs", required = false)
+    protected Integer nbBlobs;
+
+    @Param(name = "nbThreads", required = false)
+    protected Integer nbThreads = 1;
+
+    @Param(name = "logName", required = false)
+    protected String logName;
+
+    @Param(name = "logSize", required = false)
+    protected Integer logSize;
+
+    @Param(name = "logConfig", required = false)
+    protected String logConfig;
+
+    @Param(name = "basePath", required = false)
+    protected String basePath;
+
+    @OperationMethod
+    public void run() {
+        checkAccess(ctx);
+        StreamService service = Framework.getService(StreamService.class);
+        LogManager manager = service.getLogManager(getLogConfig());
+        try {
+            manager.createIfNotExists(getLogName(), getLogSize());
+            try (ProducerPool<BlobMessage> producers = new ProducerPool<>(getLogName(), manager,
+                    new FileBlobMessageProducerFactory(getListFile(), getBasePath(), getNbBlobs()),
+                    nbThreads.shortValue())) {
+                producers.start().get();
+            }
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
```

Comment:
Can you catch something more specific? At least `RuntimeException` and the needed checked exception.
Also it's probably worth rethrowing an `OperationException` instead of just logging, to inform the caller.

Reply:
I am certainly copying a bad pattern, may be this is needed to get proper HTTP error ret code I will check